--- a/qtbase/mkspecs/win32-g++/qmake.conf
+++ b/qtbase/mkspecs/win32-g++/qmake.conf
@@ -11,7 +11,7 @@
 
 MAKEFILE_GENERATOR      = MINGW
 QMAKE_PLATFORM          = win32 mingw
-CONFIG                 += debug_and_release debug_and_release_target precompile_header
+CONFIG                 += precompile_header
 DEFINES                += UNICODE
 QMAKE_COMPILER_DEFINES += __GNUC__ WIN32

@@ -30,6 +30,7 @@
 QMAKE_CFLAGS_WARN_ON    = -Wall -Wextra
 QMAKE_CFLAGS_WARN_OFF   = -w
 QMAKE_CFLAGS_RELEASE    = -O2
+QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO    = -g -O2
 QMAKE_CFLAGS_DEBUG      = -g
 QMAKE_CFLAGS_YACC       = -Wno-unused -Wno-parentheses
 QMAKE_CFLAGS_SPLIT_SECTIONS = -ffunction-sections
@@ -48,6 +49,7 @@
 QMAKE_CXXFLAGS_WARN_ON  = $$QMAKE_CFLAGS_WARN_ON
 QMAKE_CXXFLAGS_WARN_OFF = $$QMAKE_CFLAGS_WARN_OFF
 QMAKE_CXXFLAGS_RELEASE  = $$QMAKE_CFLAGS_RELEASE
+QMAKE_CXXFLAGS_RELEASE_WITH_DEBUGINFO  = $$QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO
 QMAKE_CXXFLAGS_DEBUG    = $$QMAKE_CFLAGS_DEBUG
 QMAKE_CXXFLAGS_YACC     = $$QMAKE_CFLAGS_YACC
 QMAKE_CXXFLAGS_THREAD   = $$QMAKE_CFLAGS_THREAD
