--- a/qtbase/src/gui/image/qimage.cpp	2016-02-03 15:33:28.386302991 -0800
+++ b/qtbase/src/gui/image/qimage.cpp	2016-02-03 15:34:07.682276492 -0800
@@ -4258,6 +4258,9 @@
 
 QImage QImage::smoothScaled(int w, int h) const {
     QImage src = *this;
+    if (src.isNull()) {
+        return QImage();
+    }
     switch (src.format()) {
     case QImage::Format_RGB32:
     case QImage::Format_ARGB32_Premultiplied:
@@ -4273,6 +4276,9 @@
             src = src.convertToFormat(QImage::Format_RGB32);
     }
     src = qSmoothScaleImage(src, w, h);
+    if (src.isNull()) {
+        return QImage();
+    }
     copyMetadata(src.d, d);
     return src;
 }
