--- a/qtbase/src/widgets/kernel/qwidget.cpp
+++ b/qtbase/src/widgets/kernel/qwidget.cpp
@@ -5818,7 +5818,7 @@

     pixmapOffset -= effectRect.topLeft();

-    QPixmap pixmap(effectRect.size());
+    QPixmap pixmap(effectRect.size() * m_widget->devicePixelRatio()); pixmap.setDevicePixelRatio(m_widget->devicePixelRatio());
     pixmap.fill(Qt::transparent);
     m_widget->render(&pixmap, pixmapOffset, QRegion(), QWidget::DrawChildren);
     return pixmap;
